modules/custom-status/custom-status.php

```diff
@@ -658,94 +757,54 @@ public function get_custom_statuses() {
 			return $this->custom_statuses_cache;
 		}
 
-		// Handle if the requested taxonomy doesn't exist
 		$statuses = get_terms( [
 			'taxonomy'   => self::TAXONOMY_KEY,
 			'hide_empty' => false,
+			'orderby' => 'meta_value_num',
+			'order' => 'ASC',
+			'meta_key' => self::METADATA_POSITION_KEY,
 		]);
 
 		if ( is_wp_error( $statuses ) || empty( $statuses ) ) {
 			$statuses = [];
 		}
 
-		// Expand and order the statuses
-		$ordered_statuses = [];
-		$hold_to_end      = [];
-		foreach ( $statuses as $key => $status ) {
-			// Unencode and set all of our psuedo term meta because we need the position if it exists
-			$unencoded_description = TaxonomyUtilities::get_unencoded_description( $status->description );
-			if ( is_array( $unencoded_description ) ) {
-				foreach ( $unencoded_description as $key => $value ) {
-					$status->$key = $value;
-				}
-			}
+		$statuses = array_map( [ $this, 'add_metadata_to_term' ], $statuses );
```

Comment:
I think a great alternative to `add_metadata_to_term()` would be if individual metadata fields could add themselves, e.g.

```php
$statuses = array_map( function( $status ) {
    $term_meta = apply_filters( 'vw_register_custom_status_meta', [], $status );
    $status->meta = $term_meta;
}, $statuses );
```

Then if we have individual files for metadata fields, we could have self-contained method where we add metadata for a particular field, like this for user IDs:

```php
add_filter( 'vw_register_custom_status_meta', [ __CLASS__, 'add_term_meta'] );

function add_term_meta( $term_meta, $status ) {
	$user_ids = get_term_meta( $term->term_id, self::META_KEY, true );
	if ( ! is_array( $user_ids ) ) {
		$user_ids = [];
	}

	$term_meta[ self::META_KEY ] = $user_ids;

	return $term_meta;
}
```

This would give us an easier way to add new metadata without continuing to grow this function and isolate individual metadata handling. What do you think?